### ISSUE TITLE
calling `std::mem::drop` on a `Copy`-struct should give a warning

### ISSUE THREAD

**Issue by skeleten** [rust-lang/rust#29078]
Currently [code like this][playpen-int]
```rust
fn main() {
    let i = 3;
    for _ in 0..10 {
        println!("{}", i);
        std::mem::drop(i);
    }
}
```
compiles without any warnings. 
What's happening, is that a copy of the object gets passed into std::mem::drop. I'd doubt this could be the intention of anyone writing this, so having a warning about this would be nice.
Please note that immutable borrows are `Copy` as well, so [this example][playpen-struct]
```rust
#[derive(Debug)]
struct Foo;

fn main() {
    let u = Foo;
    let i = &u;
    for _ in 0..10 {
        println!("{:?}", i);
        std::mem::drop(i);
    }
}
```
shows the same behaviour.

[playpen-int]: http://is.gd/cICPIS
[playpen-struct]: http://is.gd/fszCEI

**Comment by arielb1** [comment]
`drop` is defined simply as

```Rust
fn drop<T>(_x: T) {}
```

I think clippy already has a lint for this.

**Comment by skeleten** [comment]
I know how drop is defined, and why the code examples are both syntactly as semantically correct. I'd just really wish for an warning in such cases, as they are pretty much useless, and therefore very unlikely to be the desired behaviour.
I wasn't aware of rust-clippy yet though.

**Comment by sanxiyn** [comment]
@arielb1, I don't see this implemented in Clippy. What is it called?

**Comment by arielb1** [comment]
@sanxiyn 

Then it doesn't. Anyway, please report this as a bug to Clippy.
